#162030496 configure project

MandelaK opened the pull request:
#### What does this PR do?
Adds configuration files to repository

#### Description of Task to be completed?
Configuration files are necessary for our application to work as intended. This PR populates all files with their configurations so we can begin developing our application

#### How should this be manually tested?
This PR adds no testable functionality, but it ensures configuration errors don't occur during application setup.

#### What are the relevant pivotal tracker stories?
#162030496

MandelaK committed:
chore(configuration): configure project files

update requirements.txt

[starts #162030496]

MandelaK committed:
chore(configuration): update README

update README to include brief documentation and functionality of V2 of SendIT API

[#162030496]

MandelaK committed:
chore(configuration): set up travis

set up travis in order to implement continuous integration for my project

[starts #162030496]

MandelaK committed:
chore(setup): define create_app() function

define create_app() function for our project so it can be used by other modules

[#162030496]

MandelaK committed:
chore(configuration): configure settings and register version 2 of API

register version 2 of API
add api folder
configure __init__.py files for our api

[#162030496]

MandelaK committed:
chore(configuration): set up configuration file

set up configuration file for our project

[#162030496]

MandelaK committed:
chore(configuration): set up database configuration

add db_config.py that sets up configuration for the database

[#162030496]

MandelaK committed:
Update requirements.txt

kaka-ruto commented on .travis.yml:
```diff
@@ -0,0 +1,16 @@
+
+python:
+  - "3.6"
+  - "3.6-dev"
```
Remove 3.6-dev, let it run only on Python 3.6

kaka-ruto commented on .travis.yml:
```diff
@@ -0,0 +1,16 @@
+  - pip install -r requirements.txt
+  - pip install coveralls
+  - pip install pytest-cov
+
```
Also install postgres on travis, then set travis to create one test database and run the tests on it

kaka-ruto commented on README.md:
```diff
@@ -1,2 +1,79 @@
+
+**To install this repository**
+- clone it - $ git clone https://github.com/MandelaK/SendITAPI-V2
+- create a virtual environment 
```
Add an example, eg `Create a virtual env, for example virtualenv venv`. Then include the command to start the virtual env, eg `Run source venv/bin/activate to fire up the virtual env`

MandelaK replied:
Does this mean that I track my virtual env or simply that I add instructions on how to create and activate the virtual environment?

kaka-ruto replied:
Just give instructions

kaka-ruto commented on db_config.py:
```diff
@@ -0,0 +1,60 @@
+import psycopg2
+
+db_url = "dbname='sendit_db' user='postgres' password='postgres' host='localhost' port='5432'"
```
You never want to store sensitive information, like your database password, in your source code. If your source code is ever seen by anyone, they now have access to your database.
You should conceal/hide your production password in an environment variable.

Also create three databases, a test db, a development db and a production db, best without a password (I prefer you only set up a password for your production db).

Then configure your local machine to work with the development db, like this `export FLASK_ENV=development`

MandelaK replied:
Okay, let me see how I can do this

MandelaK committed:
update travis to create postgress test db

MandelaK committed:
chore(feedback): rename V2 folder to v2

rename V2 folder to (lowercase) v2 to handle import errors

[#162030496]

MandelaK committed:
chore(feedback): update db_config file

update db_config file to pass db_url from environment variables thus hiding our password

[#162030496]

MandelaK committed:
Update .travis.yml

MandelaK committed:
Update .travis.yml

MandelaK committed:
update travis to create test database environment

MandelaK committed:
chore(feedback): change project structure and untrack tests

change project structure and remove tracking from tests folder to test if travis build passes

[#162030496]

MandelaK committed:
Merge branch 'ch-configure-project-162030496' of https://github.com/MandelaK/SendITAPI-V2 into ch-configure-project-162030496

MandelaK committed:
chore(feedback): update requirements.txt

update requirements.txt

[#162030496]

MandelaK committed:
chore(feedback): remove tracking for tests

remove tracking for tests to see if travis build passes

[#162030496]

MandelaK committed:
update travis to use nosetest

MandelaK committed:
chore(feedback): add env examples

add .env.example file that shows users how to configure their environment variables

MandelaK committed:
chore(feedback): update readme

update readme file to include more detailed instructions on creating a virtual environment
